### PR TITLE
🎉 Google Analytics connector is failing

### LINKED ISSUE #3355: Google Analytics Connector is failing
## Expected Behavior
Job needs to fail

## Current Behavior
1. The job doesn't fail. 
In the webapp connections tab it shows as a succeeded sync, In the sync history it says running, 20 mins after the last log entry.
The connection test was successful. 
1. The job received some fields that weren't in the catalog.json (error)

## Logs
[logs-3-0.txt](https://github.com/airbytehq/airbyte/files/6456005/logs-3-0.txt)

## Steps to Reproduce
1. Build locally ./gradlew :airbyte-integrations:connectors:source-googleanalytics:build
2. Add config from lastpass to secrets/config.json
3. Run python main_dev.py read --config secrets/config.json --catalog sample_files/configured_catalog.json
 
## Severity of the bug for you
 Medium 

## Airbyte Version
 0.22.2-alpha

## Connector Version (if applicable)
0.2.3

## Additional context
GCP (BQ and compute engine for airbyte)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368231377484) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by po3na4skld**
## What
It solves #3355.

## How
Added filter into the GoogleAnalyticsSingerSource._get_reports_file_path so that it could remove non used in catalog provided by user. Also removed duplicated "title" key in spec.json file. 

## Pre-merge Checklist
- [x] *Run integration tests*
- [x] *Publish Docker images*

## Recommended reading order
1. source.py
2. spec.json

### CODE REVIEW

**Review by yevhenii-ldv — APPROVED** [review]
LGTM

File: airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py [hunk 1]
```diff
@@ -59,19 +61,26 @@ def _validate_custom_reports(self, custom_reports_data: List[dict]):
                 error_messages.append(error.message)
             raise Exception("An error occurred during custom_reports data validation: " + "; ".join(error_messages))
 
+    def read_catalog(self, catalog_path: str) -> ConfiguredAirbyteCatalog:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py:64]: this method is not called anywhere?

**Review by sherifnada — DISMISSED** [review]

File: airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py (hunk 1 above)
Comment (po3na4skld) [reply airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py:64]: It is called when performing read operation, I redefined this function from a base class to save and then filter out streams we want to use

**Review by yevhenii-ldv — CHANGES_REQUESTED** [review]
See above in comments

File: airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py (hunk 1 above)
Comment (po3na4skld) [reply airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py:64]: I got it

Comment (po3na4skld) [reply airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py:64]: @yevhenii-ldv for now it works as before

File: airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py [hunk 2]
```diff
@@ -59,19 +61,28 @@ def _validate_custom_reports(self, custom_reports_data: List[dict]):
...
+        catalog = ConfiguredAirbyteCatalog.parse_obj(self.read_config(catalog_path))
+        if not self.reports_to_read:
+            self.reports_to_read = [i.stream.name for i in catalog.streams]
+        return catalog_path
+
     def _get_reports_file_path(self, temp_dir: str, custom_reports_data: List[dict]) -> str:
         report_definition = (
             json.loads(pkgutil.get_data("tap_google_analytics", "defaults/default_report_definition.json")) + custom_reports_data
         )
+        if self.reports_to_read is not None:
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py:74]: `if self.reports_to_read`

**Review by yevhenii-ldv — APPROVED** [review]
LGTM, one small comment

**Comment by keu** [comment]
@po3na4skld I see some strange files in PR that don't belong here, could you please merge with master and ensure there are no extra files.

**Comment by po3na4skld** [comment]
> @po3na4skld I see some strange files in PR that don't belong here, could you please merge with master and ensure there are no extra files.

I think it is because I did chmod 777 on airbyte folder and after gradlew format they were added to the commits

**Review by keu — CHANGES_REQUESTED** [review]
please use this snippet to rollback changes
https://gist.github.com/liitfr/86c77bf0ea4a627ab9b62ad30c32c251

**Review by keu — APPROVED** [review]
lgtm

**Comment by davinchia** [comment]
Approach seems sound.

Two questions:
1) @keu is it possible to add the SAT to this? we should have regression guarantees.
2) this seems to fix the issue #2 in the linked issue. does this also get issue #1?

**Comment by po3na4skld** [comment]
> Approach seems sound.
> 
> Two questions:
> 
> 1. I don't see any tests here. @keu is it possible to add the SAT to this? not having any tests makes me nervous. we should have tests for regression guarantees.
> 2. this seems to fix the issue #2 in the linked issue. does this also get issue #1?

1. I will add a unit test to check this behaviour
2. can't say about #2, but #1 is probably fixed

**Comment by po3na4skld** [comment]
@davinchia 
I added tests, but couldn't run them as the API tells that Daily Request Limit is reached and throws 429 HTTP error. The same when I run the read operation. I suppose I need to move this into blocked

**Comment by davinchia** [comment]
thanks @po3na4skld 

@sherifnada is this related to https://github.com/airbytehq/airbyte/issues/3958? Have we tried getting our limit increased?

**Comment by po3na4skld** [comment]
> thanks @po3na4skld
> 
> @sherifnada is this related to #3958? Have we tried getting our limit increased?

@davinchia As I know for now we disabled any GA testing when merging to master in order to keep rate limit on an acceptable level for testing this PR

**Review by davinchia — APPROVED** [review]
@po3na4skld thanks for adding the test!

since we can't test this; I think it's fine to merge if you feel confident about it. we should make sure we  eventually come back and turn it the tests on so we have confidence things are working well.

File: airbyte-integrations/connectors/source-googleanalytics-singer/build.gradle [hunk 3]
```diff
@@ -10,12 +10,6 @@ airbytePython {
 
 
 dependencies {
-    implementation files(project(':airbyte-integrations:bases:base-standard-source-test-file').airbyteDocker.outputs)
     implementation files(project(':airbyte-integrations:bases:base-singer').airbyteDocker.outputs)
```
Comment (keu) [inline airbyte-integrations/connectors/source-googleanalytics-singer/build.gradle:13]: we don't need singer docker, right?

Comment (keu) [reply airbyte-integrations/connectors/source-googleanalytics-singer/build.gradle:13]: ok, I think we can update this connector with the CDK package at least, so we will not use base docker images.

**Pull request merged**